### PR TITLE
fetch: better typechecking fix

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

This is a better version of #15381, and addresses feedback from there.

Also reverts #15381.

### CODE REVIEW

**Review by Bo98 — APPROVED** [review]
Thanks!

**Review by MikeMcQuaid — APPROVED** [review]

File: Library/Homebrew/fetch.rb [hunk 1]
```diff
@@ -9,7 +9,7 @@ def fetch_bottle?(formula, args:)
       bottle = formula.bottle
 
       return true if args.force_bottle? && bottle.present?
-      return true if args.bottle_tag.present? && formula.bottled?(args.bottle_tag)
+      return true if args.bottle_tag.present? && formula.bottled?(args.bottle_tag&.to_sym)
```
[the remark this answers is not in the document]
Comment (carlocab) [reply Library/Homebrew/fetch.rb]: Unfortunately `brew typecheck` does not like the `&.` omission. We can fix it by extracting `args.bottle_tag` to a variable, but I feel like it makes this less readable.

[the remark this answers is not in the document]
Comment (MikeMcQuaid) [reply Library/Homebrew/fetch.rb:12]: `return true if args.bottle_tag.present? && formula.bottled?(T.must(args.bottle_tag).to_sym)`

would be the most Sorbet way to do this which basically means "I know `args.bottle_tag` will never be `nil` but, if it is, please blow up"

[the remark this answers is not in the document]
Comment (reitermarkus) [reply Library/Homebrew/fetch.rb:12]: > would be the most Sorbet way

I'd argue not having to use `T.must` in the first place is the “most Sorbet way”.

[the remark this answers is not in the document]
Comment (MikeMcQuaid) [reply Library/Homebrew/fetch.rb:12]: @reitermarkus Yes, fair point, a `nil?` check may be better in these cases for that reason.

[the remark this answers is not in the document]
Comment (Bo98) [reply Library/Homebrew/fetch.rb:12]: IMO we shouldn't be passing `args` outside of cmd files anyway.

[the remark this answers is not in the document]
Comment (carlocab) [reply Library/Homebrew/fetch.rb:12]: #15391

**Pull request merged**